commands/debug.js

```diff
@@ -1,140 +1,324 @@
+  // Debug options that allow pinging a player
+  var messagePing = message.mentions.members.first();
+  var targetID = messagePing ? message.guild.id.concat(messagePing.id) : userid;
+  if(client.userMap.has(targetID))
   {
-    max = client.cache(client.charcall.allData(client, targetID, charid, "rung"));
+    charid = client.userMap.get(targetID,"possess");
   }
-  else if((!args[1] || args[1].toLowerCase()!="confirm") && (!args[2] || args[2].toLowerCase()!="confirm"))
+  else
   {
+    message.channel.send(`That person isn't registered!`);
+    return;
+  }
+
+  switch(args[0]){
+    case "capgrist": {
+      var gristCheck = client.charcall.allData(client,targetID,charid,"grist");
+      if(gristCheck == "NONE")
+      {
+        message.channel.send(`That person isn't currently possessing anything that has grist.`);
+        return;
+      }
+
+      const gristTypes = ["build","uranium","amethyst","garnet","iron","marble","chalk","shale","cobalt","ruby","caulk","tar","amber","artifact","zillium","diamond"];
+      let max = 0;
+      if((client.charcall.allData(client, targetID, charid, "godtier") != true) && (client.charcall.allData(client, targetID, charid, "rung") != "NONE"))
+      {
+        max = client.cache(client.charcall.allData(client, targetID, charid, "rung"));
+      }
+      else if((!args[1] || args[1].toLowerCase()!="confirm") && (!args[2] || args[2].toLowerCase()!="confirm"))
+      {
         message.channel.send(`I can only help that player by setting their invalid grist levels to 0. If you're sure, add a 'confirm' to that.`);
         return;
-  }
+      }
+
+      for(i=0;i<gristTypes.length;i++){
+      if(isNaN(gristCheck[i])||gristCheck[i]===null||gristCheck[i]<0||(max > 0 && gristCheck[i]>max))
+        gristCheck[i] = max;
+      }
+      client.charcall.setAnyData(client,targetID,charid,gristCheck,"grist");
+      message.channel.send("Fixed. Probably.");
+      return;
+    }
+    break;
+
+    case "boons": {
+      let currentBoons = client.charcall.allData(client,targetID,charid,"b");
 
-  for(i=0;i<gristTypes.length;i++){
-	if(isNaN(gristCheck[i])||gristCheck[i]===null||gristCheck[i]<0||(max > 0 && gristCheck[i]>max))
-		gristCheck[i] = max;
+      if(currentBoons == "NONE")
+      {
+        message.channel.send(`That character doesn't have a PORKHOLLOW!`);
+        return;
+      }
+
+      if(args[1] && !isNaN(parseInt(args[1],10)))
+      {
+        currentBoons += parseInt(args[1],10);
+      }
+      else if(args[2] && !isNaN(parseInt(args[2], 10)))
+      {
+        currentBoons += parseInt(args[2],10);
+      }
+      else{
+        message.channel.send(`You need to provide a numerical amount of boondollars to add.`);
+        return;
+      }
+      client.charcall.setAnyData(client,targetID,charid,currentBoons,"b");
+      message.channel.send(`Your boondollars are now ${currentBoons}.`);
```

Comment:
Maybe add a conditional here to change the message to the target person if this command is used with a ping, for readability.